validate-cert.js

```diff
@@ -6,10 +6,18 @@ module.exports = function validate(pem_cert) {
   try {
     var cert = pki.certificateFromPem(pem_cert)
 
-    // check that the domain echo-api.amazon.com is present in the Subject
-    // Alternative Names (SANs) section of the signing certificate
-    if (cert.subject.getField('CN').value.indexOf('echo-api.amazon.com') === -1) {
-      return 'subjectAltName Check Failed'
+    // check that cert has a Subject Alternative Names (SANs) section
+    var altNameExt = cert.getExtension("subjectAltName")
+    if (!altNameExt) {
+      return 'invalid certificate validity (subjectAltName extension not present)'
+    }
+
+    // check that the domain echo-api.amazon.com is present in SANs section
+    var domainExists = altNameExt.altNames.some(function(name) {
+      return name.value.indexOf('echo-api.amazon.com') >= 0
```

Comment:
The docs say the domain must be "present in the `Subject Alternative Names` (SANs) section of the signing certificate". I assume present means an exact match, but it is a bit vague.

The cert I'm getting from Alexa has that exact string ('echo-api.amazon.com') in the SAN section. I'll update to an exact match if you think that's the way to go.